components/ui/Sidebar.tsx

```diff
@@ -60,21 +62,28 @@ export const Sidebar = () => {
       <aside className="h-screen w-72" aria-label="Sidebar">
         <div className="h-full overflow-y-auto bg-gray-900 py-4 px-3 dark:border-r dark:border-slate-700 dark:bg-black ">
           <div className="mt-1 flex h-full flex-col justify-between">
-            <ul className="space-y-2">
-              {tabs.map((tab) => (
-                <li key={tab.href}>
-                  <a
-                    href={tab.href}
-                    className="flex items-center rounded-lg p-3 text-base font-normal text-white hover:bg-gray-700"
-                  >
-                    {tab.icon}
-                    <span className="ml-3 flex-1 whitespace-nowrap text-sm">
-                      {tab.name}
-                    </span>
-                  </a>
-                </li>
-              ))}
-            </ul>
+            {user && (
+              <ul className="space-y-2">
+                {tabs.map((tab) => {
+                  if (tab.restrictions.includes(user.role_name)) {
+                    return (
+                      <li key={tab.href}>
+                        <a
+                          href={tab.href}
+                          className="flex items-center rounded-lg p-3 text-base font-normal text-white hover:bg-gray-700"
+                        >
+                          {tab.icon}
+                          <span className="ml-3 flex-1 whitespace-nowrap text-sm">
+                            {tab.name}
+                          </span>
+                        </a>
+                      </li>
+                    );
+                  }
+                  return null;
```

Comment:
This is definitely a nitpick but I prefer having early returns first in general, it makes reading all the possible states faster/simpler to my eyes